### PR TITLE
Add protos for directory level sandbox api

### PR DESCRIPTION
**Pull request by azliu0**
## Describe your changes

This pr adds protos for the directory-level api in WRK-521.

<details> <summary>Backward/forward compatibility checks</summary>

---

Check these boxes or delete any item (or this section) if not relevant for this PR.

- [x] Client+Server: this change is compatible with old servers
- [x] Client forward compatibility: this change ensures client can accept data intended for later versions of itself

Note on protobuf: protobuf message changes in one place may have impact to
multiple entities (client, server, worker, database). See points above.

---

</details>

### CODE REVIEW

File: modal_proto/api.proto [hunk 1]
```diff
@@ -772,6 +772,15 @@ message ContainerFileFlushRequest {
   string file_descriptor = 1;
 }
 
+message ContainerFileLsRequest {
+  string path = 1;
+}
+
+message ContainerFileMkdirRequest {
+  string path = 1;
+  bool parents = 2;
```
Comment (pawalt) [inline modal_proto/api.proto:781]: maybe `make_parents` would be more obvious

**Review by pawalt — APPROVED** [review]
LGTM

File: modal_proto/api.proto (hunk 1 above)
Comment (azliu0) [reply modal_proto/api.proto:781]: The naming mostly comes from here: https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir. 
Does that work? Don't feel too strongly about this, as it's a proto anyways

Comment (pawalt) [reply modal_proto/api.proto:781]: I don't feel strongly either - this is fine; it's just internal terminology that isn't obvious to the reader IMO

**Pull request merged**